095-stretch-cluster.md

```diff
@@ -0,0 +1,318 @@
+
+The prototype does not yet have a mechanism for controlled cleanup of remote resources when a user deliberately deletes the Kafka and KafkaNodePool CRs from the central cluster. Currently, when a user deletes these CRs, all related resources in the central cluster are removed, but remote cluster resources remain. Finding a way to handle this cleanup while avoiding unintended deletions is an open challenge.
+
+#### Rack Awareness in Stretch Clusters
+We conducted initial experiments to enable rack awareness in a Stretch Kafka cluster, ensuring that replicas are distributed across multiple Kubernetes clusters based on topology labels. This approach improves fault tolerance by leveraging Kubernetes zone-aware scheduling or manually assigned zone labels for non-zone-aware clusters
+
+For detailed implementation steps and test results, refer to our prototype [documentation](https://aswinayyolath.github.io/stretch-kafka-docs/Setting-up-Rack-Awareness-In-Stretch-Cluster/)
+
+#### Disaster Recovery - Handling Central Cluster Failure
+
+In a stretch Kafka deployment, the central Kubernetes cluster manages Kafka resources across multiple clusters, but Kafka brokers and controllers continue operating even if the central cluster fails.
+We have performed some testing on the central cluster failure scenario, which is documented [here](https://aswinayyolath.github.io/stretch-kafka-docs/Testing-cluster-failover/).
+The key challenge is restoring the administrative control plane while ensuring minimal downtime for Kafka clients.
+
+##### Recovery Approaches
+There are several possible recovery methods
+
+###### 1. Manual Recovery (Baseline Approach)
+Users bring up a new Kubernetes cluster and manually apply the Kafka, KafkaNodePool CRs.
+Alternatively, an existing remote cluster can be promoted to central by applying these CRs and updating the Cluster Operator deployment.
+This approach is simple and reliable but requires user intervention.
+
+###### 2. GitOps-Based Recovery
+
+Users store CRs in GitOps tools (e.g., ArgoCD, Flux).
+When a new central cluster is created, GitOps tools automatically reconcile the resources.
+This reduces manual effort but requires GitOps workflows to be set up in advance.
+
+#### Entity operator
+We would recommend that all KafkaTopic and KafkaUser resources are managed from the cluster that holds Kafka and KafkaNodePool resources, and that should be the cluster where the entity operator should be enabled.
+This will maintain all resource management from a central point.
+The expectation is that the entity operator will not be impacted by changes made in the prototype, but this currently requires verification.
+
+Other aspects requiring consideration and research:
```

Comment:
Does it mean that the following will not be addressed in this proposal, or just you need more time to think and propose how to handle them?

Reply:
This section was added to highlight some areas that might need attention at a later stage. Having revisited this, we believe this section should be removed and details for any specific changes we are proposing in any of these areas included in the main proposal.